<div style="text-align: center;">
  <img src="https://github.com/Hack-io-Data/Imagenes/blob/main/01-LogosHackio/logo_naranja@4x.png?raw=true" alt="esquema" />
</div>

In [148]:
import pandas as pd
import numpy as np

pd.set_option('display.max_columns', False)

# Laboratorio Pandas (EDA y Unión de Datos)

## Los datos

Para este laboratorio, trabajarás con los siguientes conjuntos de datos:

- **`netflix_originals.csv`**: Contiene información sobre producciones originales de Netflix. Las columnas que nos encontraremos en este conjunto de datos son: 

   - `Title`: Contiene el nombre del título de la producción original de Netflix.

   - `Genre`: Especifica el género o combinación de géneros de la producción.

   - `Premiere`: Indica la fecha de estreno de la producción en Netflix.

   - `Runtime`: Contiene la duración de la producción en minutos.

   - `IMDB Score`: Representa la calificación promedio que la producción ha recibido en la plataforma IMDB.

   - `Language`: Especifica el idioma principal o los idiomas en los que se presenta la producción.

- **`netflix_titles.csv`**: Contiene información sobre una variedad de títulos disponibles en la plataforma Netflix, incluyendo tanto películas como series de televisión. Las columnas que nos encontraremos en este conjunto de datos son:

   - `show_id`: Identificador único para cada título en el dataset.

   - `type`: Especifica el tipo de contenido, ya sea "Movie" (película) o "TV Show" (serie de televisión).

   - `title`: Contiene el nombre del título de la producción.

   - `director`: Muestra el nombre del director del título. Puede estar vacío en algunos casos.

   - `cast`: Lista de actores y actrices que participaron en el título. También puede estar vacío en algunos casos.

   - `country`: País o países donde se produjo el título. Esta columna también puede tener valores faltantes.

   - `date_added`: Fecha en la que el título fue añadido al catálogo de Netflix.

   - `release_year`: Año en que el título fue lanzado o producido.

   - `rating`: Clasificación del contenido en términos de audiencia (por ejemplo, PG-13, TV-MA).

   - `duration`: Duración del título. Para películas, se expresa en minutos (e.g., "90 min") y para series en temporadas (e.g., "2 Seasons").

   - `listed_in`: Categorías o géneros bajo los cuales el título está clasificado (por ejemplo, "Documentaries", "TV Dramas").

   - `description`: Breve sinopsis o descripción del contenido del título.

A lo largo de los laboratorios de Pandas, trabajarás con estos conjuntos de datos (o con los derivados del Laboratorio), algunas de las actividades que realizaremos son:

- Exploración de datos y análisis exploratorio (EDA) para familiarizarse con los conjuntos de datos.

- Realización de uniones entre conjuntos de datos, aplicando diferentes técnicas de *merging* y analizando la relevancia de cada método.

- Uso de funciones de selección y filtrado (`loc`, `iloc`) para extraer información clave.

- Desarrollo de análisis descriptivos y visualización de patrones para comprender mejor las decisiones estratégicas de la plataforma.

----

## Objetivo General de la Serie de Laboratorios



1. **Unión de `netflix_titles.csv` con `netflix_originals.csv`:**

   - **Análisis de contenido original:**  Identificarás los títulos que son producciones originales de Netflix y los compararás con los títulos adquiridos. Este análisis te permitirá:

     - Comparar la diversidad de géneros entre los títulos originales y no originales.

     - Evaluar si ciertos géneros o tipos de contenido (como documentales o series) son más frecuentes en las producciones originales.

     - Investigar si la producción original se concentra en determinados países o idiomas.


   - **Evaluación de contenido:** En esta etapa, explorarás si las producciones originales de Netflix tienden a recibir mejores o peores evaluaciones en comparación con el contenido adquirido. Algunas preguntas que se buscarán responder incluyen:

     - ¿Los títulos originales tienen una calificación promedio superior en plataformas como IMDB?

     - ¿Son las producciones originales más consistentes en términos de calidad percibida?

     - ¿Existen diferencias en la popularidad de los títulos originales según su género o país de origen?


   - **Identificación de tendencias:**  Por último, este análisis te permitirá explorar cómo ha evolucionado la estrategia de contenido de Netflix a lo largo del tiempo. Algunos puntos clave a investigar serán:

     - ¿Cómo ha crecido la producción original en comparación con la adquisición de contenido?

     - ¿Qué géneros o tipos de contenido han recibido mayor inversión en los últimos años?
     
     - ¿Existen patrones en los países o idiomas de las producciones originales, especialmente en relación con la expansión global de Netflix?


----


## Ejercicio: Análisis y Enriquecimiento del Catálogo de Netflix

### Parte 1: Análisis Exploratorio de Datos (EDA)

1. **Carga de los datos:**

   - Deberás cargar los dos conjuntos de datos que te hemos proporcionado para realizar este Laboratorio:

     - `netflix_originals.csv`

     - `netflix_titles.csv`

   Deberás revisar la estructura de cada conjunto de datos utilizando los métodos aprendidos durante la lección.

2. **Análisis de columnas y consistencia:**

   - Deberás hacer un análisis exploratorio de cada uno de los conjuntos de datos. Será obligatorio que incluyas las conclusiones generales obtenidas de cada uno de los análisis exploratorios en el Jupyter Notebook de tus soluciones.

   - Explorar qué columnas contiene cada conjunto de datos y cuáles son sus tipos de datos. Identificar si hay valores nulos o duplicados.

   - Algunas preguntas que nos podemos plantear en esta fase son (recuerda que esto es solo un ejemplo, y que podemos explorar todo lo que queramos/necesitemos para entender lo mejor posible los conjuntos de datos):

      - ¿Cuáles son los tipos de datos de cada columna? Nos puede ayudar a detectar si algún tipo de dato está incorrectamente asignado, como fechas almacenadas como cadenas de texto.

      - ¿Existen valores nulos? ¿En qué columnas están? Este paso es crucial para decidir si deben ser eliminados, rellenados o ignorados.

      - ¿Cuántos títulos únicos hay en cada conjunto de datos? Nos puede permitir verificar si existen títulos duplicados que puedan distorsionar el análisis.

      - ¿Qué géneros tenemos en el conjunto de datos? Esto nos puede ayudar a identificar los géneros más populares o incluso a identificar una posible limpieza de dicha columna para poder extraer conclusiones más claras en nuestro análisis.

### Parte 2: Unión de Datos

1. **Realización de la unión:**

   - Deberás realizar la unión que consideres más apropiada entre `netflix_titles.csv` y `netflix_originals.csv`. Además, deberás justificar el tipo de unión que has realizado.

   - Esto nos ayudará a contestar preguntas más adelente del tipo:

     - ¿Cuántos títulos en `netflix_titles.csv` son producciones originales de Netflix?

     - ¿Hay diferencias significativas en la puntuación de IMDB entre títulos originales y no originales?

### Parte 3: Uso de `loc` e `iloc`

Usando el DataFrame creado en la Parte 2: 

1. **Filtrado de datos con `loc`:**

   - Selecciona todas las filas donde el tipo de contenido sea “Movie”.

   - Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.

   - Selecciona todas las filas donde el país sea “United States”.

   - Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.

   - Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.

   - Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).

2. **Acceso a datos específicos con `iloc`:**

   - Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.

   - Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.

   - Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 

   - Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.

   - Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.

   - Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.


**NOTA IMPORTANTE**: Antes de terminar el lab, guarda el conjunto de datos creado en la fase de unión para trabajar con el en las siguientes lecciones. 

<h1 style="color: #D62828;"><u>Leyenda Colores</u></h1>

<h3 style="color: #ff8fab;">TEXTO (titulo paso)</h3>
<h5 style="color: #BEE9E8;">TEXTO  (conclusion)</h5>

<h3 style="color: #ff8fab;">Importamos los datos</h3>

In [149]:
netflix_originals  = pd.read_csv('../datos/netflix_originals.csv')
netflix_titles  = pd.read_csv('../datos/netflix_titles.csv')

<h3 style="color: #ff8fab;">Eliminamos columna 'Unnamed: 0' ya que no nos sirve</h3>

In [150]:

#! NETFLIX ORIGINALS

netflix_originals.drop(columns=["Unnamed: 0"], inplace=True)
netflix_originals.head()

Title                  Genre  ... IMDB Score          Language
0  Enter the Anime            Documentary  ...        2.5  English/Japanese
1      Dark Forces               Thriller  ...        2.6           Spanish
2          The App  Science fiction/Drama  ...        2.6           Italian
3   The Open House        Horror thriller  ...        3.2           English
4      Kaali Khuhi                Mystery  ...        3.4             Hindi

[5 rows x 6 columns]

<h3 style="color: #ff8fab;">Vemos los tipos de datos que tenemos</h3>

In [151]:

#! NETFLIX ORIGINALS

netflix_originals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 513 entries, 0 to 512
Data columns (total 6 columns):
 #   Column      Non-Null Count  Dtype  
---  ------      --------------  -----  
 0   Title       513 non-null    object 
 1   Genre       513 non-null    object 
 2   Premiere    513 non-null    object 
 3   Runtime     513 non-null    int64  
 4   IMDB Score  513 non-null    float64
 5   Language    513 non-null    object 
dtypes: float64(1), int64(1), object(4)
memory usage: 24.2+ KB


<h5 style="color: #BEE9E8;">En principio los tipos de datos parecen correctos</h5>

<h3 style="color: #ff8fab;">Comprobamos nulos y duplicados</h3>

In [152]:

#! NETFLIX ORIGINALS

#? Vemos los duplicados y nulos podamos tener
netflix_originals_dupli = netflix_originals.duplicated().value_counts()
print("Netflix Originals DUPLICADOS\n")
print(netflix_originals_dupli)

netflix_originals_null = netflix_originals.isnull().sum()
print("\n\nNetflix Originals NULOS\n\n")
print(netflix_originals_null)

netflix_originals_dupli_anios_titulos = netflix_originals.duplicated(subset=['Title', 'Premiere']).value_counts()
print("\n\nDuplicados Netflix Originals mirando Anios y Titulos\n\n")
print(netflix_originals_dupli_anios_titulos)

netflix_originals.shape

Netflix Originals DUPLICADOS

False    513
Name: count, dtype: int64


Netflix Originals NULOS


Title         0
Genre         0
Premiere      0
Runtime       0
IMDB Score    0
Language      0
dtype: int64


Duplicados Netflix Originals mirando Anios y Titulos


False    513
Name: count, dtype: int64


(513, 6)

<h5 style="color: #BEE9E8;">Vemos que la variable <b>netflix_originals</b> no tiene ni duplicados ni nulos</h5>
<h5 style="color: #BEE9E8;">Contamos con 513 titulos unicos</h5>

<h3 style="color: #ff8fab;">Identificamos las variables 'numericas' y 'categoricas' que tenemos en principio</h3>

In [153]:

#! NETFLIX ORIGINALS

#? Vemos las variables 'categoricas' y 'numericas' que tenemos en principio.
netflix_originals_num = netflix_originals.select_dtypes(np.number).head()
netflix_originals_cat = netflix_originals.select_dtypes(object).head()

print("Variables 'numericas'\n\n",netflix_originals_num)
print("\n\nVariables 'categoricas'\n\n",netflix_originals_cat)

Variables 'numericas'

    Runtime  IMDB Score
0       58         2.5
1       81         2.6
2       79         2.6
3       94         3.2
4       90         3.4


Variables 'categoricas'

              Title                  Genre           Premiere          Language
0  Enter the Anime            Documentary     August 5, 2019  English/Japanese
1      Dark Forces               Thriller    August 21, 2020           Spanish
2          The App  Science fiction/Drama  December 26, 2019           Italian
3   The Open House        Horror thriller   January 19, 2018           English
4      Kaali Khuhi                Mystery   October 30, 2020             Hindi


<h3 style="color: #ff8fab;">Vamos a hacer los mismos pasos para nuestra otra variable <b>netflix_titles</b></h3>

In [154]:

#! NETFLIX TITLES
netflix_titles.drop(columns=["Unnamed: 0"], inplace=True)
netflix_titles.head()

show_id  ...                                        description
0      s1  ...  As her father nears the end of his life, filmm...
1      s2  ...  After crossing paths at a party, a Cape Town t...
2      s3  ...  To protect his family from a powerful drug lor...
3      s4  ...  Feuds, flirtations and toilet talk go down amo...
4      s5  ...  In a city of coaching centers known to train I...

[5 rows x 12 columns]

<h5 style="color: #BEE9E8;">Eliminamos la columna <i>Unnamed: 0</i> ya que aqui tampoco nos sirve</h5>

<h3 style="color: #ff8fab;">Vemos los tipos de datos que tenemos</h3>

In [155]:
netflix_titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      3994 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


<h5 style="color: #BEE9E8;">Todo parece correcto.</h5>

<h3 style="color: #ff8fab;">
Comprobamos nulos y duplicados
</h3>

In [156]:


netflix_titles_nul = netflix_titles.isnull().sum()

#! llevarme este pequeno bloque a un archivo de funciones propias para poder reutilizarlo
df_netflix_titles_nul = pd.DataFrame(netflix_titles_nul)
df_netflix_titles_nul['%'] = round(netflix_titles.isnull().sum() / netflix_titles.shape[0] * 100, 2)

print("Netflix titles NULOS\n")
print(df_netflix_titles_nul)

netflix_titles_dup = netflix_titles.duplicated().value_counts()
print("\nNetflix titles DUPLICADOS\n")
print(netflix_titles_dup)


Netflix titles NULOS

                 0      %
show_id          0   0.00
type             0   0.00
title            0   0.00
director      2634  29.91
cast           825   9.37
country        831   9.44
date_added      10   0.11
release_year     0   0.00
rating           4   0.05
duration      4813  54.65
listed_in        0   0.00
description      0   0.00

Netflix titles DUPLICADOS

False    8807
Name: count, dtype: int64


<h5 style="color: #BEE9E8;">
Vemos que <u>no tenemos duplicados</u>, sin embargo, <u>tenemos bastantes duplicados</u>
</h5>
<h5 style="color: #BEE9E8;">
Tenemos 8807 titulos unicos
</h5>

<h3 style="color: #ff8fab;">
<u><b>PROCEDEMOS A LA UNION DE DATOS</b></u>
</h3>

In [157]:
df_catalogo = netflix_titles.merge(netflix_originals, left_on="title", right_on='Title', how='outer')
df_catalogo.shape

(8807, 18)

<h5 style="color: #BEE9E8;">
Al no tener titulos duplicados, he decidido unir por el titulo
</h5>

In [158]:
df_catalogo

show_id     type  ... IMDB Score Language
0      s2037    Movie  ...        NaN      NaN
1      s2305    Movie  ...        NaN      NaN
2      s2482    Movie  ...        NaN      NaN
3      s2325    Movie  ...        NaN      NaN
4      s5974    Movie  ...        NaN      NaN
...      ...      ...  ...        ...      ...
8802   s6178  TV Show  ...        NaN      NaN
8803   s4915  TV Show  ...        NaN      NaN
8804   s7102  TV Show  ...        NaN      NaN
8805   s5023    Movie  ...        NaN      NaN
8806   s7109    Movie  ...        NaN      NaN

[8807 rows x 18 columns]

<h5 style="color: #BEE9E8;">
Hemos hecho la unión (aunque no se si de forma correcta) y ahora toca eliminar algunas columnas ya que hay información duplicada
</h5>

<h3 style="color: #ff8fab;">
Eliminamos columnas 'sobrantes'
</h3>

In [159]:
df_catalogo.drop(columns=['Title', 'Genre', 'Premiere', 'Runtime'], inplace=True)
df_catalogo.head()

show_id   type  ... IMDB Score Language
0   s2037  Movie  ...        NaN      NaN
1   s2305  Movie  ...        NaN      NaN
2   s2482  Movie  ...        NaN      NaN
3   s2325  Movie  ...        NaN      NaN
4   s5974  Movie  ...        NaN      NaN

[5 rows x 14 columns]

<h5 style="color: #BEE9E8;">
Eliminamos 'Title', 'Genre', 'Premiere', 'Runtime' y tambien nos damos cuenta que las variables <i>Runtime</i> y <i>duration</i> deberian tener el mismo valor (al menos en las peliculas) y no lo estan haciendo.
</h5>

<h3 style="color: #ff8fab;">
Comprobamos cual es el genero que mas hay
</h3>

In [160]:
df_catalogo['listed_in'].value_counts()

listed_in
Dramas, International Movies                                    362
Documentaries                                                   359
Stand-Up Comedy                                                 334
Comedies, Dramas, International Movies                          274
Dramas, Independent Movies, International Movies                252
                                                               ... 
International TV Shows, Spanish-Language TV Shows, TV Horror      1
Children & Family Movies, Faith & Spirituality                    1
Comedies, Music & Musicals, Sports Movies                         1
Cult Movies, Dramas, Thrillers                                    1
Cult Movies, Dramas, Music & Musicals                             1
Name: count, Length: 514, dtype: int64

<h5 style="color: #BEE9E8;">
El genero que mas hay es 'Documentary'
</h5>

<h2 style="color: #ff8fab;">
Parte 3 con uso de loc e iloc
</h2>
<h3 style="color: #ff8fab;">
Filtrado de datos con <b>loc</b>
</h3>


In [161]:
df_catalogo_movies = df_catalogo[df_catalogo.loc[:, 'type'] == 'Movie']

print(f"Cantidad de elementos con el tipo 'Movie' --> {len(df_catalogo_movies)}")

df_catalogo_movies.head()

Cantidad de elementos con el tipo 'Movie' --> 6131


show_id   type  ... IMDB Score Language
0   s2037  Movie  ...        NaN      NaN
1   s2305  Movie  ...        NaN      NaN
2   s2482  Movie  ...        NaN      NaN
3   s2325  Movie  ...        NaN      NaN
4   s5974  Movie  ...        NaN      NaN

[5 rows x 14 columns]

<h5 style="color: #BEE9E8;">
Podemos apreciar que justo hay 513 elementos al igual que el total que teniamos en Netflix Originals.
</h5>

<h3 style="color: #ff8fab;">
Muestra solo los títulos (title) y la duración (duration) de todos los contenidos que sean de tipo “TV Show”.
</h3>

In [162]:
df_catalogo_tv_shows = df_catalogo[df_catalogo.loc[:, 'type'] == 'TV Show']

print(f"Cantidad --> {len(df_catalogo_tv_shows)}\n")

df_catalogo_tv_shows.head()

Cantidad --> 2676



show_id     type  ... IMDB Score Language
8    s2667  TV Show  ...        NaN      NaN
14   s2141  TV Show  ...        NaN      NaN
20   s1584  TV Show  ...        NaN      NaN
22   s2817  TV Show  ...        NaN      NaN
27   s5985  TV Show  ...        NaN      NaN

[5 rows x 14 columns]

<h3 style="color: #ff8fab;">
Selecciona todas las filas donde el país sea “United States”.
</h3>

In [163]:
df_catalogo_united_states = df_catalogo[df_catalogo.loc[:,'country'] == 'United States']
print(f"Cantidad --> {len(df_catalogo_united_states)}")
df_catalogo_united_states

Cantidad --> 2818


show_id     type  ... IMDB Score Language
5      s5696    Movie  ...        NaN      NaN
8      s2667  TV Show  ...        NaN      NaN
10     s5278    Movie  ...        NaN      NaN
13     s5972    Movie  ...        NaN      NaN
14     s2141  TV Show  ...        NaN      NaN
...      ...      ...  ...        ...      ...
8770   s8803    Movie  ...        NaN      NaN
8774   s8805    Movie  ...        NaN      NaN
8777   s8806    Movie  ...        NaN      NaN
8778   s1056    Movie  ...        NaN      NaN
8785   s5978    Movie  ...        NaN      NaN

[2818 rows x 14 columns]

<h3 style="color: #ff8fab;">
Muestra todas las películas (type = “Movie”) y selecciona las columnas title y director.
</h3>

In [164]:
df_catalogo_movies.loc[:,['title','director']]

title                     director
0                            #Alive                       Cho Il
1     #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto
2                 #FriendButMarried                Rako Prijanto
3               #FriendButMarried 2                Rako Prijanto
4                             #Roxy              Michael Kennedy
...                             ...                          ...
8798             ​Maj Rati ​​Keteki            Santwana Bardoloi
8799                    ​Mayurakshi                  Atanu Ghosh
8801           ​​Kuch Bheege Alfaaz                         Onir
8805                        반드시 잡는다                Hong-seon Kim
8806            최강전사 미니특공대 : 영웅의 탄생                Young Jun Lee

[6131 rows x 2 columns]

<h3 style="color: #ff8fab;">
Selecciona los títulos (title) y géneros (listed_in) de todas las producciones lanzadas en 2018.
</h3>

In [165]:
df_titulos_generos = df_catalogo.loc[df_catalogo['release_year'] == 2018,['title','listed_in']]
df_titulos_generos

title                                          listed_in
2        #FriendButMarried      Dramas, International Movies, Romantic Movies
4                    #Roxy                          Comedies, Romantic Movies
20      100 Days My Prince  International TV Shows, Romantic TV Shows, TV ...
21    100 Days Of Solitude                Documentaries, International Movies
39              14 Cameras                           Horror Movies, Thrillers
...                    ...                                                ...
8776                   Zoo   Dramas, Independent Movies, International Movies
8795             يوم الدين   Dramas, Independent Movies, International Movies
8797          ​Goli Soda 2   Action & Adventure, Dramas, International Movies
8801  ​​Kuch Bheege Alfaaz   Dramas, Independent Movies, International Movies
8806   최강전사 미니특공대 : 영웅의 탄생                           Children & Family Movies

[1147 rows x 2 columns]

<h3 style="color: #ff8fab;">
Selecciona las filas donde la columna director esté vacía y muestra solo los títulos (title).
</h3>

In [166]:
df_director_vacio_titulo = df_catalogo.loc[df_catalogo['director'].isnull(), ['title']]
df_director_vacio_titulo.head()

title
8               #blackAF
12                   '89
14              (Un)Well
20    100 Days My Prince
21  100 Days Of Solitude

<h3 style="color: #ff8fab;">
<b>Acceso a datos específicos con `iloc`</b>
</h3>

<h3 style="color: #ff8fab;">
Selecciona las primeras 5 filas del conjunto de datos y muestra solo las columnas title, director y country.
</h3>

In [167]:
df_catalogo.head().iloc[:,[2,3,5]]

title                     director      country
0                         #Alive                       Cho Il  South Korea
1  #AnneFrank - Parallel Stories  Sabina Fedeli, Anna Migotto        Italy
2              #FriendButMarried                Rako Prijanto    Indonesia
3            #FriendButMarried 2                Rako Prijanto    Indonesia
4                          #Roxy              Michael Kennedy       Canada

<h3 style="color: #ff8fab;">
Muestra las últimas 5 filas del conjunto de datos y selecciona las columnas title y description.
</h3>

In [168]:
df_catalogo.tail().iloc[:,[2,-3]]

title                                        description
8802             忍者ハットリくん  Hailing from the mountains of Iga, Kanzo Hatto...
8803                 海的儿子  Two brothers start a new life in Singapore, wh...
8804                 마녀사냥  Four Korean celebrity men and guest stars of b...
8805              반드시 잡는다  After people in his town start turning up dead...
8806  최강전사 미니특공대 : 영웅의 탄생  Miniforce, a special task force of elite range...

<h3 style="color: #ff8fab;">
Selecciona las primeras 10 filas del dataset y muestra las columnas desde la segunda hasta la quinta (incluidas). 
</h3>

In [169]:
df_catalogo.head(10).iloc[1:6]

show_id   type  ... IMDB Score Language
1   s2305  Movie  ...        NaN      NaN
2   s2482  Movie  ...        NaN      NaN
3   s2325  Movie  ...        NaN      NaN
4   s5974  Movie  ...        NaN      NaN
5   s5696  Movie  ...        NaN      NaN

[5 rows x 14 columns]

<h3 style="color: #ff8fab;">
Muestra las últimas 7 filas del dataset y selecciona las columnas title, director, y country usando índices posicionales.
</h3>

In [170]:
df_catalogo.iloc[-8:,[2,3,5]]

title       director      country
8799                          ​Mayurakshi    Atanu Ghosh        India
8800  ​SAINT SEIYA: Knights of the Zodiac            NaN        Japan
8801                 ​​Kuch Bheege Alfaaz           Onir        India
8802                             忍者ハットリくん            NaN        Japan
8803                                 海的儿子            NaN          NaN
8804                                 마녀사냥            NaN  South Korea
8805                              반드시 잡는다  Hong-seon Kim  South Korea
8806                  최강전사 미니특공대 : 영웅의 탄생  Young Jun Lee          NaN

<h3 style="color: #ff8fab;">
Selecciona las filas desde la 15 hasta la 25 (incluidas) y muestra las primeras 4 columnas.
</h3>

In [171]:
df_catalogo.iloc[15:26,[0,1,2,3]]

show_id  ...           director
15   s5980  ...         Adam Deyoe
16   s5981  ...        Leif Tilden
17   s3393  ...          Adze Ugah
18   s5388  ...  Nicolas Brossette
19   s5982  ...    Roland Emmerich
20   s1584  ...                NaN
21   s5983  ...                NaN
22   s2817  ...                NaN
23   s5578  ...     Marcel Barrena
24   s3315  ...                NaN
25   s5984  ...      Melinda Janko

[11 rows x 4 columns]

<h3 style="color: #ff8fab;">
Muestra las filas 20 a 30 y solo las columnas title, release_year, y rating usando índices posicionales.
</h3>

In [172]:
df_catalogo.iloc[20:31,[2,7,8]]

title  release_year rating
20                        100 Days My Prince          2018  TV-14
21                      100 Days Of Solitude          2018  TV-MA
22                                100 Humans          2020  TV-14
23                                100 Meters          2016  TV-MA
24       100 Things to do Before High School          2014   TV-Y
25  100 Years: One Woman's Fight for Justice          2016  TV-14
26                                100% Halal          2020  TV-14
27                               100% Hotter          2017  TV-14
28                           1000 Rupee Note          2014  TV-14
29                              12 ROUND GUN          2017  TV-MA
30                          12 Years Promise          2014  TV-14